x/bank/app_test.go

```diff
@@ -146,22 +170,43 @@ func TestMsgSendMultipleOut(t *testing.T) {
 		Address: addr1,
 		Coins:   sdk.Coins{sdk.NewCoin("foocoin", 42)},
 	}
-
 	acc2 := &auth.BaseAccount{
 		Address: addr2,
 		Coins:   sdk.Coins{sdk.NewCoin("foocoin", 42)},
 	}
-	accs := []auth.Account{acc1, acc2}
 
-	mock.SetGenesis(mapp, accs)
+	mock.SetGenesis(mapp, []auth.Account{acc1, acc2})
 
-	// Simulate a Block
-	mock.SignCheckDeliver(t, mapp.BaseApp, []sdk.Msg{sendMsg2}, []int64{0}, []int64{0}, true, priv1)
+	testCases := []appTestCase{
+		{
+			simBlock: true,
+			msgs:     []sdk.Msg{sendMsg2},
+			accNums:  []int64{0},
+			accSeqs:  []int64{0},
+			expPass:  true,
+			privKeys: []crypto.PrivKey{priv1},
+		},
+		{
+			addr:  addr1,
```

Comment:
Any particular reason you have a switch case to check the expected state transition, instead of having a different field in the `appTestCase` struct for expected outcome? Putting it in the `appTestCase` struct seems more optimal to me.

Reply:
You mean why do I leverage the `mock#*` calls? Not particularly. This was the original code and figured I'd leave it this way to leverage existing functionality. I can take a look at those function calls and put the `require` statements directly in here if that's what you'd prefer?

Reply:
No, I mean that the struct has a "testSim" flag. When that is true, it runs a block, when its false, it tests that the expected action happened.

I think it would be easier if we had a list of expected address / coin amounts after running the tx, and we put those directly in the struct. this would mean adding a new struct type:
`expectedAddressCoinAmount {Address, Coins}` (better name probably exists)

and then adding to the `appTestCase` struct, an `[]expectedAddressCoinAmount` and removing `testSim`

Reply:
I see, yeah we can do that for sure 👍